src/main/scala/nodes/learning/KernelGenerator.scala

```diff
@@ -0,0 +1,273 @@
+package nodes.learning
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import utils._
+
+
+// Trait to be implemented by Kernel functions
+trait KernelGenerator[T] {
```

Comment:
The first element is that element dot trainingData -- i.e. With a linear kernel lets say you have a full dataset Y (m x d) and a train data set X (n x d). Then the full kernel is Y * X^T (size is m x n)

Now if I only get 1 row of Y, say y1 then the kernel is y1 * X^T -- which is of size 1 x n

Does this make sense ?